### PR TITLE
 Add shared/dynamic IoT.js module template

### PR DESCRIPTION
**Pull request by galpeter**
Introduced the shared module template.
With this a dynamically loadable IoT.js module
can be quickly developed.

The new template contains a simple readme on how
to build and test.

Usage:

./tools/iotjs-create-module.py --template shared <module_name>

### CODE REVIEW

**Comment by galpeter** [comment]
Depends on: #1531

**Comment by daeyeon** [comment]
@galpeter please get `js/test.js:1: incorrect license error` fixed.

**Comment by daeyeon** [comment]
Can we create a test for this and add it to test/run_pass?

**Comment by galpeter** [comment]
@daeyeon I think it is possible to create a test case. I'll check what can we do and create another PR or update this depending on the complexity.

**Comment by galpeter** [comment]
I've rebased the PR and also added a test case

**Comment by daeyeon** [comment]
@galpeter Thanks for the test case. Please squish two commits into a single one.

**Comment by galpeter** [comment]
@daeyeon the first commit is the other PR. This PR depends on that

**Comment by daeyeon** [comment]
@galpeter Sorry. I thought the other pr has been landed already.

**Comment by zherczeg** [comment]
@daeyeon I think the other PR is ready to land

File: test/dynamicmodule/src/module_entry.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  jerry_value_t prop_name = jerry_create_string((const jerry_char_t*)"demokey");
+  jerry_value_t prop_value = jerry_create_number(3.4);
+
+  jerry_set_property(object, prop_name, prop_value);
+
+  jerry_release_value(prop_name);
+  jerry_release_value(prop_value);
+
+
+  iotjs_jval_set_property_number(object, "counter", ++counter);
```
Comment (zherczeg) [inline test/dynamicmodule/src/module_entry.c]: One newline above.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**